core/src/Cuda/Kokkos_Cuda_Graph_Impl.hpp

```diff
@@ -117,10 +117,7 @@ struct GraphImpl<Kokkos::Cuda> {
   //  requires NodeImplPtr is a shared_ptr to specialization of GraphNodeImpl
   //  Also requires that the kernel has the graph node tag in it's policy
   void add_node(std::shared_ptr<NodeImpl> const& arg_node_ptr) {
-    static_assert(
-        NodeImpl::kernel_type::Policy::is_graph_kernel::value,
-        "Something has gone horribly wrong, but it's too complicated to "
-        "explain here.  Buy Daisy a coffee and she'll explain it to you.");
```

Comment:
I agree that this error message is not useful to the readers.
Is it feasible to replace it with some other useful message instead of completely removing it?

Reply:
I wouldn't know what else to say. We basically add this trait to kernels that should be run in a graph and copy the kernel in the process. Thus, this `static_assert` could trigger if we forget to add this trait (or pass the original kernel here).

Reply:
Sounds fair.